pkg/conditions/conditions.go

```diff
@@ -105,3 +105,13 @@ func (c Conditions) ToConditions(generation int64) []metav1.Condition {
 
 	return result
 }
+
+func (c Conditions) AllOfStatus(status metav1.ConditionStatus) bool {
```

Comment:
The name is a bit confusing. "All of..." is normally used in the sense of "all of the members", so you'd expect it to return a collection. `AllHaveStatus`? `AllStatusEqual`?